app/models/targeting.rb

```diff
@@ -0,0 +1,29 @@
+class Targeting < ActiveRecord::Base
+
+  belongs_to :user
+  belongs_to :bookmark
+
+  has_many :targeting_hosts, :dependent => :destroy
+  has_many :hosts, :through => :targeting_hosts
+  has_many :template_invocations, :dependent => :destroy
+
+  validate :bookmark_or_query
+
+  def resolve_hosts!
+    fail _("Cannot resolve hosts without a user") if user.nil?
```

Comment:
we use raise in Foreman core everywhere, I know Katello prefers fail, I wonder with what we should be consistent here or if we allow using both

anyway whatever we choose I think we should raise/fail with some custom exception